pydra/utils/hash.py

```diff
@@ -472,11 +472,29 @@ def bytes_repr_fileset(
     fileset: FileSet, cache: Cache
 ) -> Iterator[ty.Union[CacheKey, bytes]]:
     fspaths = sorted(fileset.fspaths)
+    # Yield the cache key for the fileset, which is a tuple of the file-system paths
+    # and their mtime. Is used to store persistent cache of the fileset hashes
+    # to avoid recomputation between calls
     yield CacheKey(
         tuple(repr(p) for p in fspaths)  # type: ignore[arg-type]
         + tuple(p.lstat().st_mtime_ns for p in fspaths)
     )
-    yield from fileset.__bytes_repr__(cache)
+    cls = type(fileset)
+    yield f"{cls.__module__}.{cls.__name__}:".encode()
+    for key, chunk_iter in fileset.byte_chunks():
+        yield (",'" + key + "'=").encode()
+        yield from chunk_iter
+
+
+# Need to disable the mtime cache key for mocked filesets. Used in doctests
+@register_serializer(MockMixin)
+def bytes_repr_mock_fileset(
```

Comment:
I don't like putting test code in main modules. What about adding `pydra.testing.fixtures` and including:

```
@pytest.fixture(scope='session')
def register_mock_mixin_serializer():
    @register_serializer(MockMixin)
    ...
```

Then in `conftest.py`:

```
from pydra.testing.fixtures import register_mock_mixin_serializer
```

This would allow downstream tools that need this feature to also put this import in their `conftest.py`.

Reply:
I agree with the sentiment in general, but in this case I reckon that might be creating a gotcha that would take a bit of tracking down how to solve for little practical benefit, i.e if someone creates a new task package and wants to write a doctest using `NiftiGz.mock()`, then they would need to know that they also have to add this fixture